### PR TITLE
feat: support setting language for the extension

### PR DESCRIPTION
**Pull request by KariHall619**
**What type of PR is this?**
implement AI model auto-detection and enhance plugin loading reliability

**What this PR does / why we need it**:

This PR introduces comprehensive AI plugin integration capabilities with intelligent model detection and robust plugin loading mechanisms. The changes address critical reliability issues in plugin initialization and provide a foundation for dynamic AI model management.

**Key Features:**

1. **AI Model Auto-Detection System** (`pkg/testing/model_detector.go`)
   - Implements automatic detection of available AI models across multiple providers (OpenAI, Claude, local models)
   - Eliminates the need for manual model configuration
   - Provides comprehensive error handling and logging

2. **Enhanced Plugin Loading Reliability** (`console/atest-ui/src/views/Extension.vue`)
   - Implements exponential backoff retry mechanism (up to 10 attempts with incremental delays: 50ms, 100ms, 150ms...)
   - Proper loading state management to prevent UI freezing
   - Comprehensive error handling with detailed logging
   - Optimized CSS/JS loading sequence

3. **AI Plugin Communication Infrastructure** (`pkg/testing/remote/grpc_store.go`)
   - Extends gRPC store to support AI operations (`ai.generate`, `ai.capabilities`)
   - Implements parameter encoding and response conversion for AI queries
   - Standardizes AI response format for consistent integration

4. **Configuration Optimization** (`cmd/testdata/stores.yaml`)
   - Updates AI plugin to use Unix socket communication (`unix:///tmp/atest-ext-ai.sock`)
   - Enables auto-detection with improved model description
   - Removes hardcoded model defaults to support dynamic detection

5. **Development Environment Improvements**
   - Fixed extension menu parameter passing in frontend routing
   - Updated Vite configuration to avoid port conflicts (5173 → 5174)
   - Complete unit test coverage for model detection functionality

**Why we need it:**
- **Reliability**: The original plugin loading was fragile and failed when plugins needed initialization time
- **User Experience**: Auto-detection eliminates manual configuration and reduces setup complexity  
- **Scalability**: Provides a robust foundation for supporting multiple AI providers and models
- **Maintainability**: Comprehensive error handling and logging improve debugging and monitoring

**Testing:**
- Added 91 lines of unit tests for model detection functionality
- Covers various detection scenarios and edge cases
- Ensures stability and reliability of the AI integration system

This PR transforms the AI plugin system from a basic proof-of-concept into a production-ready, intelligent, and reliable integration platform.

### CODE REVIEW

File: console/atest-ui/src/views/Extension.vue [hunk 1]
```diff
@@ -9,33 +9,55 @@ const props = defineProps<Props>()
...
+
+            // Start the retry mechanism
+            checkPluginLoad();
         });
     } catch (error) {
-        console.log(`extension load error: ${(error as Error).message}`)
-    } finally {
-        console.log('extension load finally');
+        console.log(`extension load error: ${(error as Error).message}`);
+        loading.value = false; // Set loading to false on error
```
Comment (LinuxSuRen) [inline console/atest-ui/src/views/Extension.vue:60]: I think this should be in the `finally` block.

File: console/atest-ui/src/views/Extension.vue [hunk 2]
```diff
@@ -9,33 +9,55 @@ const props = defineProps<Props>()
...
+
+                console.log(`Plugin load attempt ${retries + 1}/${maxRetries + 1}`);
+
+                if (plugin && plugin.mount) {
+                    console.log('extension load success');
+                    const container = document.getElementById("plugin-container");
+                    if (container) {
+                        container.innerHTML = ''; // Clear previous content
+                        plugin.mount(container);
+                        loading.value = false;
```
Comment (LinuxSuRen) [inline console/atest-ui/src/views/Extension.vue:38]: This is unnecessary. Only do it one time is enough.

File: console/atest-ui/src/views/Extension.vue [hunk 3]
```diff
@@ -9,33 +9,55 @@ const props = defineProps<Props>()
...
 
-            const plugin = window.ATestPlugin;
-            
-            if (plugin && plugin.mount) {
-                console.log('extension load success');
-                const container = document.getElementById("plugin-container");
-                if (container) {
-                    container.innerHTML = ''; // Clear previous content
-                    plugin.mount(container);
+            // Implement retry mechanism with exponential backoff
```
Comment (LinuxSuRen) [inline console/atest-ui/src/views/Extension.vue:26]: This is good. But I don't know when it will fail. I never meet the failure.

File: console/atest-ui/src/App.vue [hunk 4]
```diff
@@ -178,7 +178,7 @@ API.GetMenus((menus) => {
       <WelcomePage v-else-if="panelName === 'welcome' || panelName === ''" />
 
       <span v-for="menu in extensionMenus" :key="menu.index" :index="menu.index">
-        <Extension v-if="panelName === menu.index" :name="menu.name" />
+        <Extension v-if="panelName === menu.index" :name="menu.index" />
```
Comment (LinuxSuRen) [inline console/atest-ui/src/App.vue:181]: Why not name?

File: pkg/testing/model_detector.go
Comment (LinuxSuRen) [inline pkg/testing/model_detector.go:1]: As you mentioned. I prefer moving this to the AI extension. We need keep the core be clean and clear.

File: console/atest-ui/src/views/Extension.vue (hunk 1 above)
Comment (KariHall619) [reply console/atest-ui/src/views/Extension.vue:60]: You are right. loading.value = false should indeed be placed in the finally block to ensure that the operation is executed regardless of success or failure.

File: console/atest-ui/src/views/Extension.vue (hunk 2 above)
Comment (KariHall619) [reply console/atest-ui/src/views/Extension.vue:38]: 如果在其他地方也有相同代码，那确实应该只保留必要的一次.

File: console/atest-ui/src/views/Extension.vue (hunk 3 above)
Comment (KariHall619) [reply console/atest-ui/src/views/Extension.vue:26]: In some cases this may happen:
- The JavaScript file has been loaded but the global object window.ATestPlugin has not been initialized
- The resources that the plugin depends on are still being loaded asynchronously
- It takes time for the browser to execute scripts, especially large plugins.

File: console/atest-ui/src/App.vue (hunk 4 above)
Comment (LinuxSuRen) [reply console/atest-ui/src/App.vue:181]: yes, you're right.

Comment (LinuxSuRen) [reply console/atest-ui/src/App.vue:181]: yes, you're right.

File: pkg/testing/remote/grpc_store.go [hunk 5]
```diff
@@ -444,3 +452,65 @@ func (g *gRPCLoader) Close() {
...
 	}
 }
+
+// handleAIQuery handles AI-specific queries
+func (g *gRPCLoader) handleAIQuery(query map[string]string) (testing.DataResult, error) {
+	method := query["method"]
+
+	var dataQuery *server.DataQuery
+	switch method {
+	case "ai.generate":
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply pkg/testing/remote/grpc_store.go:462]: But the way, there are not many places to ref `ai.generate`. So, maybe we don't need a constant of it.

File: console/atest-ui/src/views/Extension.vue [hunk 6]
```diff
@@ -1,48 +1,83 @@
...
+            const checkPluginLoad = (retries = 0, maxRetries = 10) => {
+                const globalScope = globalThis as { ATestPlugin?: { mount?: (el: Element, context: PluginAppContext) => void } };
+                const plugin = globalScope.ATestPlugin;
+
+                if (plugin && plugin.mount) {
+                    const container = document.getElementById("plugin-container");
+                    if (container) {
+                        container.innerHTML = ''; // Clear previous content
+                        plugin.mount(container, pluginContext);
```
Comment (LinuxSuRen) [inline console/atest-ui/src/views/Extension.vue]: 这行代码会导致函数不兼容，已有插件全都是一个参数

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

**Review by LinuxSuRen — APPROVED** [review]
